include/deal.II/fe/fe_poly.templates.h

```diff
@@ -251,19 +256,26 @@ fill_fe_values (const Mapping<dim,spacedim>                                  &ma
 
       if (flags & update_hessians && cell_similarity != CellSimilarity::translation)
         {
-          // compute the hessians in the unit cell (accounting for the Jacobian gradiant)
-          for (unsigned int i=0; i<quadrature.size(); ++i)
-            {
-              fe_data.untransformed_shape_hessians[i] = fe_data.shape_hessians[k][i];
-            }
-
-          correct_untransformed_hessians (fe_data.untransformed_shape_hessians,
-                                          mapping_data, output_data, quadrature.size(), k);
-
-          mapping.transform (fe_data.untransformed_shape_hessians,
+          mapping.transform (fe_data.shape_hessians[k],
                              mapping_covariant_gradient,
                              mapping_internal,
                              output_data.shape_hessians[k]);
+
+          for (unsigned int i=0; i<quadrature.size(); ++i)
+            for (unsigned int m=0; m<spacedim; ++m)
+              output_data.shape_hessians[k][i] -=
+                mapping_data.jacobian_pushed_forward_grads[i][m]
+                * output_data.shape_gradients[k][i][m];
```

Comment:
It may be hiding from me in plain sight, but I can't see an operator* for rank-3 tensors and rank-1 ones. I've toyed briefly with the idea of a general one-rank-contraction operator* for tensor along the lines:

```
template <int dim, typename Number>
template <int order1, int order2>
Tensor<order1+order2-2,dim,Number>
operator * (const Tensor<order1,dim,Number> &src1,
            const Tensor<order2,dim,Number> &src2)
{
  Tensor<order1+order2-2,dim,Number> dest;
  for (unsigned int i=0; i<dim; ++i)
    {
      dest[i] = src1[i] * src2;
    }
  return dest;
}
```

But it wasn't as straight forward to get it to compile as I'd hoped, so I shelved the idea. Any advice on how I can get something like this to work?